share/new_eds/validation.go

```diff
@@ -0,0 +1,64 @@
+package eds
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/celestiaorg/rsmt2d"
+
+	"github.com/celestiaorg/celestia-node/share"
+	"github.com/celestiaorg/celestia-node/share/shwap"
+)
+
+var _ Accessor = validation{}
```

Comment:
Should implement Closer as we discussed.

Reply:
Added WithCloser(Accessor,Closer) AccessorCloser. Very handy and will be reused later

Reply:
Hmm, ok